tutorials/tts/TTS_pronounciation_customization.ipynb

```diff
@@ -0,0 +1,290 @@
+    "import torch\n",
+    "\n",
+    "# Load mel spectrogram generator\n",
+    "spec_generator = nemo_tts.models.FastPitchModel.from_pretrained(\"tts_en_fastpitch_ipa\").eval()\n",
+    "# Load vocoder\n",
+    "vocoder = nemo_tts.models.HifiGanModel.from_pretrained(model_name=\"tts_hifigan\").eval()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# to use dictionary entries for known words\n",
+    "spec_generator.vocab.phoneme_probability = 1\n",
+    "spec_generator.vocab.g2p.phoneme_probability = 1\n",
+    "\n",
+    "# helper fucntion to generate audio\n",
+    "def generate_audio(input_text):\n",
+    "    parsed = spec_generator.parse(input_text)\n",
+    "    spectrogram = spec_generator.generate_spectrogram(tokens=parsed)\n",
+    "    audio = vocoder.convert_spectrogram_to_audio(spec=spectrogram)\n",
+    "    display(ipd.Audio(audio.detach().to('cpu').numpy(), rate=22050))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "text = \"paracetamol can help reduce fever.\"\n",
+    "generate_audio(text)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "During preprocessing, unambiguous dictionary words are converted to phonemes, while OOV and words with multiple entries are kept as graphemes. For example, **paracetamol** is missing in from the phonetic dictionary, and **can** has 2 forms."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(f\"Input before tokenization: |{' '.join(spec_generator.vocab.g2p(text))}|\\n\")\n",
+    "for word in [\"paracetamol\", \"can\"]:\n",
+    "    word = word.upper()\n",
+    "    phoneme_forms = spec_generator.vocab.g2p.phoneme_dict[word]\n",
+    "    print(f\"Number of phoneme forms for wordPhoneme forms for '{word}': {len(phoneme_forms)} -- {phoneme_forms}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Input customization\n",
+    "\n",
+    "One a one-time customization suffice, one can pass phonemes directly as input to the model.\n",
+    "Let's replace the word **paracetamol** with the desired phonetic form in our text example."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(f\"Original text: {text}\")\n",
+    "\n",
+    "new_pronunciation = \"ˌpæɹəˈsitəmɔl\"\n",
+    "phoneme_form = \"\".join([f\"|{s}|\" for s in new_pronunciation])\n",
+    "text_with_phonemes = text.replace(\"paracetamol\", phoneme_form)\n",
+    "print(f\"Text with phonemes: {text_with_phonemes}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "generate_audio(text_with_phonemes)"
```

Comment:
Can we upload sample audio? Something like: https://github.dev/nvidia-riva/tutorials/blob/main/tts-python-basics-and-customization-with-ssml.ipynb

Reply:
done